### PR TITLE
Fix Schema Registry 4xx handling and add tests for negative scenarios

### PR DESCRIPTION
**Pull request by arerlend**
Also - I am aware that blocking async calls is not the best behavior.  This will be fixed when CachedSchemaRegistryClient and core serializer and deserializer implementations are converted to use Reactor types.

### CODE REVIEW

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java [hunk 1]
```diff
@@ -118,13 +122,26 @@ public SchemaRegistryObject register(
             "Registering schema. Group: '{}', name: '{}', serialization type: '{}', payload: '{}'",
                 schemaGroup, schemaName, schemaType, schemaString);
 
-        SchemaId schemaId;
+        CreateSchemaResponse response;
         try {
-            schemaId = this.restService.createSchema(schemaGroup, schemaName, schemaType, schemaString);
+            response = this.restService.createSchemaWithResponseAsync(schemaGroup, schemaName, schemaType, schemaString)
+                .block(HTTP_REQUEST_TIMEOUT);
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java:128]: If we want to include a timeout, it must be configurable by the user?

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java [hunk 2]
```diff
@@ -171,23 +194,35 @@ public SchemaRegistryObject getSchemaByGuid(String schemaId) {
...
             return schemaStringCache.get(schemaString).getSchemaId();
         }
 
-        SchemaId schemaId;
+        GetIdBySchemaContentResponse response;
         try {
-            schemaId = this.restService.getIdBySchemaContent(schemaGroup, schemaName, schemaType, schemaString);
+            response = this.restService
+                .getIdBySchemaContentWithResponseAsync(schemaGroup, schemaName, schemaType, schemaString)
+                .block(HTTP_REQUEST_TIMEOUT);
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java:207]: Same here, timeout should be an optional method param. You can still have a default in case it's not set.

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java (hunk 1 above)
Comment (arerlend) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java:128]: #12341
